lab_3_ann_retriever/main.py

```diff
@@ -475,6 +680,28 @@ def _find_closest(self, vector: Vector, k: int = 1) -> list[tuple[float, int]] |
 
         In case of corrupt input arguments, None is returned.
         """
+        if not isinstance(vector, tuple) or not vector:
+            return None
+        stack = [(self._root, 0.0)]
+
+        while stack:
+            node, depth = stack.pop(0)
+            if node is None:
+                continue
+
+            operator = Node()
+            if operator.right_node is None and operator.left_node is None:
```

Comment:
it will be None always, no? You are creating an instance before it

Reply:
it is not about operator, it was another referencing the current result. In this case it is definetely strange, but i have done something nice - i deleted this operator) but in general None is the indicator of the completeness of the division into branches in KDTree, i think i can`t live without it

Reply:
I hope that you can live without this code...